### DISCUSSION TITLE
Hope for bricked Solos/Pros?

### DISCUSSION THREAD

**Discussion by GerryDazoo** [GerryDazoo/Slinger#343] [General]
I stuck my bricked Solo back on the interwebs and did a packet capture of what was going on.
Starting at packet 217.
[Bricked_solo.zip](https://github.com/GerryDazoo/Slinger/files/10067012/Bricked_solo.zip)
So there is some interesting stuff here.
As some have surmised the Sling back-office servers are still on-line. I think all they really shutdown were the servers that supported setting up a video streaming session. 
In the packet capture are all the DNS queries the SB is doing, Ignore the ones that end in ht.home
First off is myip.sling.com. You'll see that it resolves to a host on the akamai.net CDN. In my case a1093.d.akamai.net.
So if you going to block stuff by IP you can't rely on what others have posted. Do a ping yourself to see what it resolves to on your network and hope it doesn't change.
You will also see that the Solo is going out and touching the servers every 4 hours. So Sling is still keeping track of where your SB is on the WAN and the info also includes the SB's IP address on you LAN and Finderid. 

Part of that process includes sending FW info to analytics.sling.com and getting a response. This is why I have a little hope. Maybe it will be possible to spoof the response from Sling to give it different FW info and get the Solo to download it. Fingers crossed but this is why we need a real packet capture  of the bricking process to see what Sling is sending.

Another interesting point is the TCP connection to myip.sling.com never drops (tcp Keep-alives). I'm not saying that the FW supports this but it's "theoretically" possible for Sling to tunnel back into the box over this open TCP connection to do whatever they want. Maybe this is how they are being bricked? Who knows, yet another reason to get a packet capture of the bricking process.

A note on traffic blocking. analytics.sling.com resolves to a server(s) on AWS. In my case p-dany-analytics-1643478883.us-west-2.elb.amazonaws.com. I believe the elb refers to "enhanced load balancer".  So again if you're blocking by IP you need to get the proper IPs from where you sit.  At my house that domain name actually resolves to 4 IP addresses  52.32.51.203,   52.38.155.127,   44.235.167.139 and  35.155.185.12.

Here's also my take on trying to block by domain name. It depends on the router implementation. To be reliable it would have to monitor all the DNS traffic going into the LAN to make sure it knows the current IPs to block. A broken implementation would only resolve the IP once when you setup the outgoing firewall rules. The other possibility is the blocking relies on having all the DNS traffic on the LAN using the DNS server built into the router, to make it easier to monitor the DNS info. So beware if you've setup your router or SB to use a third party DNS service like Mr. Google 8.8.8.8 which would bypass the router's internal DNS server.

If your router supports blocking out-going traffic by MAC address, that I think that will be the most reliable. It is often called "parental controls" so you can stop the kids from surfing the web after bedtime.  On my router it's called "Device Filtering". In any event I think most routers have this capability buried somewhere. Read the manual.

**Comment by pfuller88** [comment]
Great news.   It would be nice to revive the solo.

**Reply by GerryDazoo** [reply]
Note sure how great it is. I don't have a ProHD to get a "bricking" packet capture, or a solo or pro to sacrifice. I'm relying on the kindness of strangers to help out.

**Reply by theweatherguy** [reply]
Gerry,  over the weekend, I discovered that I have a slingbox solo that hasn't been plugged in for years.  I know it's not bricked.  (In fact I can't seem to find the cables or the power adaptor.)   I stopped using it about 4 or 5 years ago when I went to the two slingbox 350s that I use now.  
I'm happy to donate it to you "for science" and as a thank you for all your incredible work.    How can I send it to you?

**Reply by GerryDazoo** [reply]
Send it to

**Reply by theweatherguy** [reply]
It's been boxed  and it will be sent to you via UPS tomorrow morning.    I'll send you the UPS tracking information slip privately when I get it.

**Reply by theweatherguy** [reply]
Gerry,
My USPS tracking shows that the unbricked Slingbox Solo I sent you was delivered to you today.   Did you receive it?  I don't seem to have any way of contacting you directly.

**Reply by GerryDazoo** [reply]
Not, here. I have an issue with UPS to. It said my ProHD package was delivered to a place in Kentucky. Turn out it's just gone into the black-hole of Canadian customs.

**Reply by theweatherguy** [reply]
I don't know if you received the tracking number and scan of the receipt I sent via direct email to GerryDazoo <notifications@github.com>   Looking at it,I guess that's not a direct address, is it (?)  Is there a direct way of contacting you here?

I'd like to send you a scanned copy of the receipt and customs paperwork privately so you might track it down.    If you didn't receive it, I need to know....it was insured....plus it cost me $43 USD  to send it to you.   I've always had a very positive view of you Canadians...surely you guys have a mail service that works.     (BTW, that was my X-mas gift to you as a thank you for all your efforts).

**Reply by GerryDazoo** [reply]
email me 
The notification thing should have worked. But it may have ended up in my spam folder and I missed it,

**Reply by theweatherguy** [reply]
Thanks.  I resent the email I sent to you with an attached receipt and tracking info to the above email address.  Please let me know if you receive the tracking info.

**Comment by rbf121** [comment]
From what I remember reading, it attempts to update the firmware every 24-48 hours and upon reboot.  Maybe if you do a packet capture as you reboot the device, you might see where it goes out for firmware updates.  

I don't have a solo or prohd but these service URLs still list the current firmware as the latest:
http://services.sling.com/configservices/rest/v2/configuration/product/SOLO/version/2.1.270.json
http://services.sling.com/configservices/rest/v2/configuration/product/PROHD/version/2.1.420.json

**Reply by GerryDazoo** [reply]
That packet capture was after a restart and it lasted 3 days. So unless the bricked Solo will respond to  a different message sent back by Sling that's all I got. This is why I need a packet capture of a "bricking" to see what Sling sends back to the SB.

**Reply by kasibarndoor** [reply]
On eBay in the states there is a Slingbox solo for $7.16 the seller accepts offers, the postage to the UK is $29 so perhaps someone on the other side of the pond could snap it up and run a test on it

**Reply by kasibarndoor** [reply]
sorry missing information is the bay id is 224493467893

**Reply by GerryDazoo** [reply]
Thanks for the info. I need something that can be unbicked to start with.
I'm also very concerned now people might be putting up bricked boxes on ebay. All the description says is it powers up.

**Comment by floormanager1** [comment]
Happy to get one ini the UK unbrick it and send it on.  There's one for £199!  Not sure I will go that high but I'll keep an eye out on some others which are £20

**Reply by GerryDazoo** [reply]
No need I have one on the way.

**Reply by Chaskel** [reply]
@GerryDazoo once you get the Solo unit, please feel free to post the capture file of the firmware update traffic that takes place and I can see if I may be able to assist. I suspect it would happen on initial power up, but there's always a chance it could happen later than that.

**Comment by RandomMan719** [comment]
@GerryDazoo 
With the help of an 8 hour wire capture that @proudx2 did on his ProHD, I have been able to block my ProHD from being bricked, and access it remotely through the Internet.  I actually just did a remote digital scan of the builtin Tuner on the ProHD, with Mac Slingplayer 1.0.10 and direct ip alias.  I also watched the stream using Slinger remotely for about a half an hour, with no issues.

His capture showed that his ProHD uses these specific ports to talk to the outside world, ports: (80,443,3074,3490,3724,5678,5680,8080)

So I made a firewall rule in pfsense, that blocks my ProHD from using any of those ports as destination ports.  And I can use any other port to access it remotely. (I will be trying to setup rules on my remote routers, that are TP-Links, hopefully I can use the same method)

PS 
I bricked the ProHD 4 or 5 times yesterday getting my firewall rules setup correctly.  Good thing, it only takes a few minutes to unbrick and try again.  I would not recommend anyone try this with the Solo or the Pro, until we know if they use the same ports, to talk to the outside world.

(I made a couple edits removed port 490, typo, and added port 5680)

**Reply by gbjbasdw** [reply]
Can we get a pdf of how to unbrick a ProHD please. I did read something today but it sounded quite difficult to me.

**Reply by RandomMan719** [reply]
@gbjbasdw   It's right here!

Slingbox ProHD PC Unbricking-Setup-Network-Guide #357

Mac Unbricking ProHD Guide is here, but not in PDF form:
ProHD Macintosh Revival Guide #314

**Reply by Toei79** [reply]
Whats the prohd

I have one sb300

**Comment by pfuller88** [comment]
Any more information on the solo or PRO model progress, if any?

**Reply by RandomMan719** [reply]
Not that I am aware of.   I have 2 solos and a pro, that are unbricked.  I would like to do some testing on them, but want to make sure I don't brick them in the process.  I will post more, if I figure anything out.

**Reply by GerryDazoo** [reply]
I got nothin'. We  have isolated the message that bricks the ProHDs. But it's contents are encrypted. Do you know any cryptographers? I think we know the encryption algorithm and we have an idea what part of the plaintext may be. All we need to do is figure out the 128 bit key. But even if we do figure it out we would still need to figure out how the solo/pros responded to it. I don't have a warm feeling that there is anything we can do to revive them.

**Reply by Chaskel** [reply]
Any chance you can share the complete pcap file so I can take a look?

**Reply by RandomMan719** [reply]
@Chaskel I will just post it here, It's on another computer, I will go grab it.

**Reply by Chaskel** [reply]
I'm afraid not (have never had a need for it). I can try to create an account later today (possibly in ~4hrs). If you are RandomMan719 there, I'll reach out to you once I sign up.

**Reply by RandomMan719** [reply]
@Chaskel  here are a couple packet captures, I did on the ProHD, using pfsense builtin packet capture.  Let us know, if you figure anything out, Thanks!

**Reply by Chaskel** [reply]
Thanks I'm taking a look now. I also just signed up for Discord so feel free to give me whatever information needed there.

EDIT: I just looked at the captures but there isn't really much there. If I would be able to get a packet capture of a non-bricked Solo and Pro-HD booting up, that would be great.

**Reply by RandomMan719** [reply]
@Chaskel I didn't see that you joined the Slinger Discord Server, here is a link to it.

Slinger Discord Server > https://discord.gg/rMsMepD3c5 #154

**Reply by Chaskel** [reply]
Thanks @RandomMan719. I've just sent some thoughts Gerry's way for review.
